### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by kelvin8773**
* add Calculator logic
* add Operation logic
* add display logic

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+const Calculate = data => {
+  const { numOne, operation, numTwo } = data;
+
+  if (!numTwo) return numOne;
+
+  return Operate(numOne, numTwo, operation);
+};
+
+export default Calculate;
```
Comment (shloch) [inline src/logic/calculate.js]: this function should have 2 parameters
- data object (you already got that) and
- button name (you dont have this)
[[see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/buttonAsArgument.png)]

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      break;
+    case '-':
+      result = one.minus(two).toString();
+      break;
+    case 'x':
+      result = one.times(two).toString();
+      break;
+    case '÷':
+      if (two.eq(0)) {
+        result = ' -N/A- ';
```
Comment (shloch) [inline src/logic/operate.js]: great thing you managed this

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import Big from 'big-js';
+
+const Operate = (numOne, numTwo, operation) => {
+  Big.DP = 10;
+  Big.RM = 1;
+  Big.NE = -5;
+  Big.PE = 10;
```
Comment (shloch) [inline src/logic/operate.js]: what is this for ?? (just curious) 😊

Comment (kelvin8773) [reply src/logic/operate.js]: @shloch  it is for restricting how many numbers can be returned, more detail you can check here - 
http://mikemcl.github.io/big.js/#dp

File: src/components/App.js [hunk 4]
```diff
@@ -16,61 +15,7 @@ class App extends React.Component {
...
-        case '=':
-        default:
-      }
-    }
-
-    result = Calculate({ numOne, operation, numTwo });
-    this.setState({
-      numOne, operation, numTwo, result,
-    });
+    this.setState(prevState => Calculate(prevState, button));
```
Comment (shloch) [inline src/components/App.js]: nice

**Pull request merged**